Extension/src/LanguageServer/client.ts

```diff
@@ -864,6 +876,122 @@ export class DefaultClient implements Client {
+                if (paths[i].split("\\").pop() !== undefined) {
+                    option = paths[i].split("\\").pop();
+                    isCompiler = true;
+                }
+            }
+            if (option !== undefined && isCompiler) {
+                const path: string | undefined = paths[i].replace(option, "");
+                items.push({ label: option, description: localize("found.string", "Found at {0}", path), index: i });
+            } else {
+                items.push({ label: paths[i], index: i });
+            }
+        }
+
+        const selection: IndexableQuickPickItem | undefined = await vscode.window.showQuickPick(items, options);
+        return (selection) ? selection.index : -1;
+    }
+
+    public async handleCompilerQuickPick(): Promise<void> {
+        const settings: OtherSettings = new OtherSettings();
+        let paths: string[] = [];
+        if (compilerDefaults.knownCompilers !== undefined) {
+            paths = compilerDefaults.knownCompilers.map(function (a: configs.KnownCompiler): string { return a.path; });
+        }
+        paths.push(localize("selectAnotherCompiler.string", "Select another compiler on my machine"));
+        paths.push(localize("installCompiler.string", "Help me install a compiler"));
+        paths.push(localize("noConfig.string", "Do not configure a compiler (not recommended)"));
+        const index: number = await this.showSelectCompiler(paths);
+        if (index === paths.length - 1) {
+            paths[index] = "";
+            settings.defaultCompiler = "";
+        } else if (index === paths.length - 2) {
+            switch (os.platform()) {
+                case 'win32':
+                    vscode.commands.executeCommand('vscode.open', "https://go.microsoft.com/fwlink/?linkid=2217614");
+                    return;
+                case 'darwin':
+                    vscode.commands.executeCommand('vscode.open', "https://go.microsoft.com/fwlink/?linkid=2217706");
+                    return;
+                default: // Linux
+                    vscode.commands.executeCommand('vscode.open', "https://go.microsoft.com/fwlink/?linkid=2217615");
+                    return;
+            }
+        } else if (index === paths.length - 3) {
+            const result: vscode.Uri[] | undefined = await vscode.window.showOpenDialog();
+            if (result !== undefined && result.length > 0) {
+                util.addTrustedCompiler(compilerPaths, result[0].fsPath);
+                settings.defaultCompiler = result[0].fsPath;
+                compilerDefaults = await this.requestCompiler(compilerPaths);
+                this.updateClientConfigurations();
+                return;
+            }
+        } else {
+            if (index !== -1) {
+                util.addTrustedCompiler(compilerPaths, paths[index]);
+            }
+        }
+        // If a compiler is selected, update the default.compilerPath user setting.
+        if (index < paths.length - 3) {
```

Comment:
You should check index !== -1 here too.

Reply:
Actually, can you just check for index === -1 earlier on and exit from the function?